scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/RevioPacBioProcessor.java

```diff
@@ -0,0 +1,413 @@
+      new RevioPacBioProcessor.ProcessMetadata[] {
+        processString("//RunDetails/TimeStampedName", PacBioNotificationDto::setRunAlias),
+        processString(
+            "//CollectionMetadata/@InstrumentName", PacBioNotificationDto::setSequencerName),
+        processDate("//Run/@WhenStarted", PacBioNotificationDto::setStartDate),
+        processSampleInformation()
+      };
+
+  /**
+   * Extract a PacBio-formatted string from the metadata file and put the parsed result into the
+   * DTO.
+   *
+   * @param expression the XPath expression yielding the date
+   * @param setter the writer for the date
+   */
+  private static RevioPacBioProcessor.ProcessMetadata processDate(
+      String expression, BiConsumer<PacBioNotificationDto, Instant> setter) {
+    XPathExpression expr = RunProcessor.compileXPath(expression)[0];
+    return (document, dto, timeZone) -> {
+      String date = (String) expr.evaluate(document, XPathConstants.STRING);
+      if (date != null) {
+        DateTimeFormatter formatter = DateTimeFormatter.ISO_DATE_TIME;
+        ZonedDateTime parsedDateTime = ZonedDateTime.parse(date, formatter);
+        setter.accept(dto, parsedDateTime.toInstant());
+      }
+    };
+  }
+
+  /**
+   * Extract a number from the metadata file and put the result into the DTO.
+   *
+   * @param expression the XPath expression yielding the number
+   * @param setter the writer for the number
+   */
+  private static RevioPacBioProcessor.ProcessMetadata processNumber(
+      String expression, BiConsumer<PacBioNotificationDto, Double> setter) {
+    XPathExpression expr = RunProcessor.compileXPath(expression)[0];
+    return (document, dto, timeZone) -> {
+      Double result = (Double) expr.evaluate(document, XPathConstants.NUMBER);
+      if (result != null) {
+        setter.accept(dto, result);
+      }
+    };
+  }
+
+  // Revio Samples
+  private static RevioPacBioProcessor.ProcessMetadata processSampleInformation() {
+    XPathExpression[] expr =
+        RunProcessor.compileXPath(
+            "//ResultsFolder",
+            "//SubreadSet/@UniqueId",
+            "//WellSample/@Name",
+            "//AutomationParameters/AutomationParameter[@Name='MovieLength']/@SimpleValue");
+    return (document, dto, timeZone) -> {
+      String position =
+          StringUtils.substringBetween(
+              (String) expr[0].evaluate(document, XPathConstants.STRING), "/", "/");
+      String containerSerialNumber = (String) expr[1].evaluate(document, XPathConstants.STRING);
+      String poolName = (String) expr[2].evaluate(document, XPathConstants.STRING);
+      String movieLength = (String) expr[3].evaluate(document, XPathConstants.STRING);
+
+      // From DefaultPacBio, checking to make sure string is valid
+      if (isStringBlankOrNull(position)
+          || isStringBlankOrNull(containerSerialNumber)
+          || isStringBlankOrNull(poolName)
+          || isStringBlankOrNull(movieLength)) {
+        return;
```

Comment:
If you can't find one piece of information, you don't include any of the other pieces?